src/cljs/kixi/hecuba/tabs/hierarchy/tapestry.cljs

```diff
@@ -0,0 +1,385 @@
+                legend-data (->> (-> cursor :heatmap :element-legend)
+                                 (viz/svg-plot2d-cartesian)
+                                 (svg/svg {:width @chart-width :height @chart-height}))]
+            (dom/div #js {:style #js {:position "relative"
+                                      :overflow "hidden"
+                                      :whiteSpace "nowrap"
+                                      :fontFamily "sans-serif"
+                                      :fontSize "11px"
+                                      :textAlign "center"
+                                      :width (str (+ 100 @chart-width) "px")}}
+                     (dom/div nil (dom/span nil x-label))
+                     (dom/span #js {:style #js {:float "left"
+                                                ;;:height (str (/ @chart-height 2) "px")
+                                                :margin (str (/ @chart-height 2) "px -30px 0px -30px")
+
+                                                :WebkitTransform "rotate(-90deg)"
+                                                :MozTransform "rotate(-90deg)"
+                                                :msTransform "rotate(-90deg)"
+                                                :OTransform "rotate(-90deg)"
+                                                :filter "progid:DXImageTransform.Microsoft.BasicImage(rotation=3)"}} y-label)
+                     (dom/div #js {:style #js {:display "inline-block"}
+                                   :dangerouslySetInnerHTML #js
+                                   {:__html (hiccups/html main-data)}})
+                     (dom/div #js {:style #js {:display "inline-block"}
+                                   :dangerouslySetInnerHTML #js
+                                   {:__html (hiccups/html legend-data)}}))))))))
+
+(defn url-str [start end entity_id device_id type]
+  (str "/4/entities/" entity_id "/devices/" device_id "/measurements/"
+       type "?startDate=" start "&endDate=" end))
+
+(defn date->amon-timestamp [date]
+  (->> date
+       (tf/parse (tf/formatter "yyyy-MM-dd"))
+       (tf/unparse (tf/formatter "yyyy-MM-dd HH:mm:ss"))))
+
+(defn calculate-end-date [start-date]
+  (let [d (tf/parse (tf/formatter "yyyy-MM-dd") start-date)]
+    (->> (t/plus d (t/weeks 2))
+         (tf/unparse (tf/formatter "yyyy-MM-dd HH:mm:ss")))))
+
+(defn process-data
+  "This will currently only work for hourly data"
+  [data]
+  ;; step 1
+  (let [remove-times  (map #(-> %
+                                (assoc-in [:date] (.slice (:timestamp %) 0 10))
+                                (assoc-in [:time] (.slice (:timestamp %) 11 19))
+                                (dissoc :sensor_id)
+                                (dissoc :timestamp)) data)
+        grouped-data  (group-by :date remove-times)]
+    (if-not (= (count grouped-data) x-readings) ;; ensure there are 2 weeks worth
```

Comment:
i'd prefer  `(if (not= a b)` over `(if-not (= a b))` - 'cos the predicate is clearer then